src/services/helpers.ts

```diff
@@ -23,14 +23,40 @@ export class NetworkError extends Error {
   }
 }
 
+const timeoutAfter = (seconds: number): Promise<Error> => {
+  return new Promise((resolve, _reject) => {
+    setTimeout(() => {
+      resolve(new Error(`request timed-out after ${seconds} seconds`));
+    }, seconds * 1000);
+  });
+};
+
 export async function fetchWithValidator<T, O, I>(
   validator: Type<T, O, I>,
   requestInfo: RequestInfo,
   init?: RequestInit
 ): Promise<T> {
   let response: Response;
+  /*
+   * AbortController() is supported globally for Jest with Node v15 and Jest v27 and above.
+   * Thus, logic is added to check if AbortController is valid for Jest testing purpose only.
+   * TODO: Remove the logic check when node and jest version is compatible.
+   */
   try {
-    response = await fetch(requestInfo, init);
+    const controller = typeof AbortController
```

Comment:
why do we need the check here for `typeof AbortController`? can we just `const controller = new AbortController();`?

Reply:
Logic added mainly for unit-testing purpose only as AbortController() is not support globally for Node v12.22.1 and Jest v26.6.3 which cause the unit-testing to fail in [92aea07](https://github.com/rationally-app/mobile-application/pull/569/commits/92aea0710ffb30e175604da1ae550efab08751ab). AbortController() is supported globally from Node v15 and Jest v27. [reference](https://github.com/facebook/jest/pull/11182)